engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/gui/csv/CSVTableExportTask.java

```diff
@@ -66,7 +66,7 @@
    */
   private final MasterReport report;
   private StatusListener statusListener;
-
+  private Boolean createParentFolder;
```

Comment:
The upper class boolean is back. Please replace it (again) with a primitive boolean. Whereever you use simple flags, stick to the safest value possible.